### PR TITLE
8326121: vmTestbase/gc/g1/unloading/tests/unloading_keepRef_rootClass_inMemoryCompilation_keep_cl failed with Full gc happened. Test was useless.

### PR DESCRIPTION
**Pull request by walulyai**
Hi all,

Please review this change to trigger SkippedException instead of RunTimeExceptions when a test is deemed useless because of full-GCs. The test triggers frequent `System.gc()` calls which in some cases may aggressively shrink the heap, and subsequent allocations lead to full-GC. This may affect the usefulness of the test, but it is not a test failure. So we added  SkippedException to indicate this situation.

Testing: Local testing with very small heaps.
              Tier 1.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Warning
&nbsp;⚠️ Found trailing period in issue title for `8326121: vmTestbase/gc/g1/unloading/tests/unloading_keepRef_rootClass_inMemoryCompilation_keep_cl failed with Full gc happened. Test was useless.`
Found leading lowercase letter in issue title for `8326121: vmTestbase/gc/g1/unloading/tests/unloading_keepRef_rootClass_inMemoryCompilation_keep_cl failed with Full gc happened. Test was useless.`

### Issue
 * [JDK-8326121](https://bugs.openjdk.org/browse/JDK-8326121): vmTestbase/gc/g1/unloading/tests/unloading_keepRef_rootClass_inMemoryCompilation_keep_cl failed with Full gc happened. Test was useless. (**Bug** - P4)

### Reviewers
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to [59d2462c](https://git.openjdk.org/jdk/pull/19432/files/59d2462cbd15851cf69016e89eb709924e18966d)
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19432/head:pull/19432` \
`$ git checkout pull/19432`

Update a local copy of the PR: \
`$ git checkout pull/19432` \
`$ git pull https://git.openjdk.org/jdk.git pull/19432/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19432`

View PR using the GUI difftool: \
`$ git pr show -t 19432`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19432.diff">https://git.openjdk.org/jdk/pull/19432.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19432#issuecomment-2135879617)

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/share/test/Tests.java [hunk 1]
```diff
@@ -82,6 +84,8 @@ public void execute(Object o) {
                                         ((Runnable) o).run();
                                 if (o instanceof TestExitCode)
                                         exitCode = ((TestExitCode) o).getExitCode();
+                        }catch (SkippedException se) {
```
Comment (tschatzl) [inline test/hotspot/jtreg/vmTestbase/nsk/share/test/Tests.java:87]: Spacing.

File: test/hotspot/jtreg/vmTestbase/gc/g1/unloading/UnloadingTest.java [hunk 2]
```diff
@@ -161,7 +161,7 @@ private static void checkGCCounters() {
         }
 //        if (WhiteBox.getWhiteBox().g1GetTotalFullCollections() != 0 || (oldGenBean != null && oldGenBean.getCollectionCount() != 0)) {
```
Comment (tschatzl) [inline test/hotspot/jtreg/vmTestbase/gc/g1/unloading/UnloadingTest.java:162]: Maybe remove these System.out.println() here and above - your call.

**Review by tschatzl — APPROVED** [review]
Looks good, please fix the spacing problem.

Also, please update the CR with the evaluation in the PR description.

**Review by albertnetymk — APPROVED** [review]

**Comment by walulyai** [comment]
Thanks @albertnetymk and @tschatzl for the reviews!

/integrate

**Pull request closed without merging**